### ISSUE TITLE
Compatibility with Bronze mod

### ISSUE THREAD

**Issue by vico93** [ThexXTURBOXx/Expanded-Armory-Legacy#1]
Basically, if the mod detects the presence of [Bronze](https://github.com/Khazoda/bronze) ([MR](https://modrinth.com/mod/bronze)) installed, it would provide also the correspondent set of tools for that metal as well.

According to the author:

> This means that the bronze sword is halfway between iron and diamond in terms of damage, though the diamond sword has far greater durability.
> The bronze axe on the other hand is on par with iron and diamond axes in terms of damage, but has slightly better durability than iron, while having far less than diamond.

**Comment by makumaku1974** [comment]
can i provide a texture?

**Comment by ThexXTURBOXx** [comment]
> Basically, if the mod detects the presence of [Bronze](https://github.com/Khazoda/bronze) ([MR](https://modrinth.com/mod/bronze)) installed, it would provide also the correspondent set of tools for that metal as well.

Good idea - I was also already thinking about integrations with other mods, but in a more "general" way.
For older versions, I wanted to see whether the `OreDictionary` has some, e.g., "ingotBronze" entry and if that's the case, enable the corresponding weapon(s) and recipes.
For newer versions, I wanted to achieve something similar, but with tags instead.

Should be doable, but requires quite some effort.

> can i provide a texture?

Of couse you can :)
However, please take your time, since I don't know when this might/will be done!

**Comment by ThexXTURBOXx** [comment]
I have done some minor work to show how to easily integrate Bronze into BWM: Legacy: https://github.com/ThexXTURBOXx/bronze/commit/217d0039e5d417cc967454175198c2bedd315252
Currently, it only adds a boomerang entity, but the rest can be integrated the exact same way :)

One thing to note: I was stupid and had all the item attributes private...
So, mod creators need to wait for the next version of BWM: Legacy in order to *properly* integrate their mod (affects only [this cosmetic change](https://github.com/ThexXTURBOXx/bronze/commit/217d0039e5d417cc967454175198c2bedd315252#diff-5ff8895c6b1e7bc87a826df2d30eb16793ce68adce2f87081a3028086818cba3R9-R12))

When the next version (probably only a minor release this time) is released, I will notify the mod author.
You can also already do it now, if you want. Then, she/they can already work on this while I fix up this mod.
If you do contact her, please include a link to this very comment here as well, so that as much information as possible is given.

**Comment by vico93** [comment]
Pingging @Khazoda to let her be aware

**Comment by makumaku1974** [comment]
![Screenshot 2024-06-25 123543](https://github.com/ThexXTURBOXx/Balkons-WeaponMod-Legacy/assets/129804277/3359a245-43cb-4a2d-ba8a-9be83c8192f8)



> Pingging @Khazoda to let her be aware

The redesign for Bronze Battle Axe

**Comment by ThexXTURBOXx** [comment]
I think I will move this issue to a more appropriate repository :)

**Comment by khazoda** [comment]
> Pingging @Khazoda to let her be aware

Hi, thanks for the ping! Looking at the expanded armory legacy's repo it seems this mod is for 1.7.10? I feel I may be missing something, as Bronze is currently only available for 1.21 fabric.

In terms of mod integration, this is something I've tried tackling with another of my mods, [Basic Weapons](https://github.com/Khazoda/basic-weapons), where I wrote the integration in the weapon mod, rather than the bronze mod.

I'm happy to work alongside you on this though, I'm just a little lost currently seeing the repo is for Minecraft 1.7.10 😅

**Comment by makumaku1974** [comment]
> > Pingging @Khazoda to let her be aware
> 
> Hi, thanks for the ping! Looking at the expanded armory legacy's repo it seems this mod is for 1.7.10? I feel I may be missing something, as Bronze is currently only available for 1.21 fabric.
> 
> In terms of mod integration, this is something I've tried tackling with another of my mods, [Basic Weapons](https://github.com/Khazoda/basic-weapons), where I wrote the integration in the weapon mod, rather than the bronze mod.
> 
> I'm happy to work alongside you on this though, I'm just a little lost currently seeing the repo is for Minecraft 1.7.10 😅

For now, we only need for 1.21. Do not include 1.7.10 yet because it's kind of not needed right now. Since this is a very old mod and revived by @ThexXTURBOXx

**Comment by ThexXTURBOXx** [comment]
@Khazoda 
Oh yes, that might look rather confusing. I transferred this issue from https://github.com/ThexXTURBOXx/Balkons-WeaponMod-Legacy to over here!
Balkon's WeaponMod: Legacy is available for most major versions between 1.7.10 and 1.21 (and is still in active development for all of them).
I have done efforts over [here](https://github.com/ThexXTURBOXx/bronze/commit/e2d792b15da0b6c5a27bc71f5301eb795ce8e133) to show how to integrate your Bronze mod into BWM: Legacy. The same approach pretty much works analogously for other mods.
So, this is probably what you might want to focus on :)

Edit: Please also keep in mind that the boomerang is not the only weapon that you can integrate.
And, since last version, there is an even better API for item integration, which I have not yet implemented in that repo.
Basically, [these functions](https://github.com/ThexXTURBOXx/Balkons-WeaponMod-Legacy/blob/1.21/common/src/main/java/ckathode/weaponmod/WMItemBuilder.java#L39-L127) will allow you to create the exact items with the correct settings automatically.
This is probably better than doing manually [as seen here](https://github.com/ThexXTURBOXx/bronze/commit/e2d792b15da0b6c5a27bc71f5301eb795ce8e133#diff-6470680a4f40c4e5585091ef0239909cee104a106e462e1d19b51229f5eee05aR19-R20). Just replace this by the appropriate function call (in this case [`createStandardBoomerang`](https://github.com/ThexXTURBOXx/Balkons-WeaponMod-Legacy/blob/1.21/common/src/main/java/ckathode/weaponmod/WMItemBuilder.java#L61-L63)) and you will get the same result, but more cleaner and more independently!

**Comment by khazoda** [comment]
Thanks for clarifying, the version makes sense to me now.
Is your suggestion writing the integration within the bronze mod using your item builder functions, along with all the weapon textures?
If so, I'm not sure I'm up to doing that unfortunately.

I'm trying to keep my vanilla+ mods, which includes Bronze, as simple as possible because I intend to update them all very quickly when each new Minecraft version drops, and am trying to keep file sizes and code bloat down too.
The way I wrote integration with the Basic Weapons mod is by conditionally registering bronze weapons in that mod, using the ToolMaterial from Bronze.
Would something like that work for Balkon's WeaponMod, or would that work against your design principles?

**Comment by ThexXTURBOXx** [comment]
You're welcome :)
And yes, that would have been indeed my idea.
However, I do get your point. So, it might be best then to do the integration in this mod right here (Expanded Armory: Legacy).
Thank you for your replies, though! I appreciate it!

Then, this issue has to wait a bit until I have successfully ported Expanded Armory: Legacy to 1.21 (which will take quite a good amount of time again!)

**Comment by khazoda** [comment]
Alright sounds good 😊👍
I'm happy to do a PR with the integration once Expanded Armory: Legacy is ported!
